File: command/levelButton.js
const { SlashCommandBuilder, ButtonBuilder, ButtonStyle, ActionRowBuilder, MessageFlags, TextDisplayBuilder } = require('discord.js');

function setup(client, { userStats }) {
  const command = new SlashCommandBuilder().setName('level-button').setDescription('Show your level with a button');
  client.application.commands.create(command);

  client.on('interactionCreate', async interaction => {
    if (!interaction.isChatInputCommand() || interaction.commandName !== 'level-button') return;
    const stats = userStats[interaction.user.id] || { level:1 };
    const button = new ButtonBuilder().setCustomId('get-level').setLabel('Get Level').setStyle(ButtonStyle.Primary);
    const row = new ActionRowBuilder().addComponents(button);
    await interaction.reply({
      components: [
        new TextDisplayBuilder().setContent(`# Your Level\nYou are level **${stats.level}**!`),
        row,
      ],
      flags: MessageFlags.IsComponentsV2,
    });
  });

  client.on('interactionCreate', async interaction => {
    if (!interaction.isButton() || interaction.customId !== 'get-level') return;
    const stats = userStats[interaction.user.id] || { level:1 };
    await interaction.reply({
      components: [new TextDisplayBuilder().setContent(`You are level ${stats.level}`)],
      flags: MessageFlags.Ephemeral | MessageFlags.IsComponentsV2,
    });
  });
}

module.exports = { setup };

File: command/level_button.py
import discord


def setup(tree, user_stats, *_, **__):
    """Register a demo level command with a button."""

    class LevelView(discord.ui.View):
        def __init__(self, user_id: int) -> None:
            super().__init__()
            self.user_id = user_id

        @discord.ui.button(label="Get Level", style=discord.ButtonStyle.primary)
        async def get_level(
            self,
            interaction: discord.Interaction,
            button: discord.ui.Button,  # pragma: no cover - no logic depends on button
        ) -> None:
            stats = user_stats.get(self.user_id, {"level": 1})
            await interaction.response.send_message(
                f"You are level {stats['level']}", ephemeral=True
            )

    @tree.command(name="level-button", description="Show your level with a button")
    async def level_button(interaction: discord.Interaction) -> None:
        stats = user_stats.get(interaction.user.id, {"level": 1})
        embed = discord.Embed(
            title="Your Level", description=f"You are level **{stats['level']}**!"
        )
        view = LevelView(interaction.user.id)
        await interaction.response.send_message(embed=embed, view=view)
